platform/api/oci/instance.go

```diff
@@ -45,98 +50,133 @@ func (a *API) CreateInstance(name, userdata, sshKey string) (*Machine, error) {
-		a.TerminateInstance(inst.ID)
+		a.TerminateInstance(*inst.Id)
 		return nil, fmt.Errorf("listing vnic attachments: %v", err)
 	}
 
-	if len(vnicAttachments.Attachments) < 1 {
-		a.TerminateInstance(inst.ID)
+	if len(vnicAttachments.Items) < 1 {
+		a.TerminateInstance(*inst.Id)
 		return nil, fmt.Errorf("couldn't get VM information")
 	}
-	vnic, err := a.client.GetVnic(vnicAttachments.Attachments[0].VnicID)
+	vnic, err := a.vn.GetVnic(context.Background(), core.GetVnicRequest{
+		VnicId: vnicAttachments.Items[0].VnicId,
+	})
 	if err != nil {
 		return nil, fmt.Errorf("getting vnic: %v", err)
 	}
 
+	if inst.DisplayName == nil {
+		return nil, fmt.Errorf("received instance display name nil")
+	}
+
+	if vnic.Vnic.PublicIp == nil {
+		return nil, fmt.Errorf("received vnic public ip nil")
+	}
+
+	if vnic.Vnic.PrivateIp == nil {
+		return nil, fmt.Errorf("received vnic private ip nil")
+	}
+
 	return &Machine{
-		Name:             inst.DisplayName,
-		ID:               inst.ID,
-		PublicIPAddress:  vnic.PublicIPAddress,
-		PrivateIPAddress: vnic.PrivateIPAddress,
+		Name:             *inst.DisplayName,
+		ID:               *inst.Id,
+		PublicIPAddress:  *vnic.Vnic.PublicIp,
+		PrivateIPAddress: *vnic.Vnic.PrivateIp,
 	}, nil
 }
 
 func (a *API) TerminateInstance(instanceID string) error {
-	return a.client.TerminateInstance(instanceID, nil)
+	_, err := a.compute.TerminateInstance(context.Background(), core.TerminateInstanceRequest{
+		InstanceId: &instanceID,
+	})
+	return err
 }
 
 // ConsoleHistory is deleted when an instance is terminated, as such
 // we just return errors and let the history be deleted when the instance
 // is terminated.
 func (a *API) GetConsoleOutput(instanceID string) (string, error) {
-	metadata, err := a.client.CaptureConsoleHistory(instanceID, nil)
+	metadata, err := a.compute.CaptureConsoleHistory(context.Background(), core.CaptureConsoleHistoryRequest{
+		CaptureConsoleHistoryDetails: core.CaptureConsoleHistoryDetails{
+			InstanceId: &instanceID,
+		},
+	})
 	if err != nil {
 		return "", fmt.Errorf("capturing console history: %v", err)
 	}
 
-	consoleHistoryStatus, err := a.client.GetConsoleHistory(metadata.ID)
+	consoleHistoryStatus, err := a.compute.GetConsoleHistory(context.Background(), core.GetConsoleHistoryRequest{
+		InstanceConsoleHistoryId: metadata.ConsoleHistory.Id,
+	})
 	if err != nil {
 		return "", fmt.Errorf("getting console history status: %v", err)
 	}
 
 	err = util.WaitUntilReady(5*time.Minute, 10*time.Second, func() (bool, error) {
-		consoleHistoryStatus, err = a.client.GetConsoleHistory(metadata.ID)
+		consoleHistoryStatus, err := a.compute.GetConsoleHistory(context.Background(), core.GetConsoleHistoryRequest{
+			InstanceConsoleHistoryId: metadata.ConsoleHistory.Id,
+		})
 		if err != nil {
 			return false, fmt.Errorf("getting console history status: %v", err)
 		}
 
-		if consoleHistoryStatus.State != "SUCCEEDED" {
+		if consoleHistoryStatus.ConsoleHistory.LifecycleState != "SUCCEEDED" {
```

Comment:
Mabye move these strings out to consts as well, to protect against typos. I know I personally can't spell succeed to save my life.